includes/marketplace-suggestions/class-wc-marketplace-suggestions-api.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * REST API Marketplace suggestions API
+ *
+ * Handles requests for marketplace suggestions data
+ *
+ * @package WooCommerce/API
+ * @since   3.6.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Pull suggestion data from remote endpoint & cache in a transient.
+ *
+ * @return array of json API data
+ */
+function wc_marketplace_suggestions_get_api_data() {
+	$suggestion_data_url = 'https://d3t0oesq8995hv.cloudfront.net/add-ons/marketplace-suggestions.json';
+
+	$suggestion_data = get_transient( 'wc_marketplace_suggestions' );
+
+	if ( false !== $suggestion_data ) {
+		return $suggestion_data;
+	}
+
+	$raw_suggestions = wp_safe_remote_get(
+		$suggestion_data_url,
+		array( 'user-agent' => 'WooCommerce Marketplace Suggestions' )
+	);
+	if ( is_wp_error( $raw_suggestions ) ) {
+		return array();
+	}
+
+	// Parse the data to check for any errors.
+	// If it's valid, store structure in transient.
+	$suggestions = json_decode( wp_remote_retrieve_body( $raw_suggestions ) );
+	if ( $suggestions && is_array( $suggestions ) ) {
+		set_transient( 'wc_marketplace_suggestions', $suggestions, WEEK_IN_SECONDS );
+		return $suggestions;
+	}
+
+	return array();
+}
+
+/**
+ * Suggestion data GET handler.
+ */
+function wc_marketplace_suggestions_ajax_handler() {
+	$suggestion_data = wc_marketplace_suggestions_get_api_data();
+
+	echo wp_json_encode( $suggestion_data );
+
+	wp_die();
+}
+
+
+
+/**
+ * Return an array of suggestions the user has dismissed.
+ */
+function wc_marketplace_suggestions_get_dismissed() {
```

Comment:
Correct – that's by design right now.

Note that we should be able set up the json file so the dismiss slug represents a product rather than a particular suggestion instance.